2.17 Gradle Platform and Catalog

elect86 opened the pull request:
As titled, this is on the same line as Scijava

During the publication, Maven will run Gradle to generate the [Platform](https://docs.gradle.org/current/userguide/platforms.html) (Gradle counterpart for Maven BOM) and [Catalog](https://docs.gradle.org/current/userguide/version_catalogs.html) 

![image](https://github.com/user-attachments/assets/0063d216-65fe-4c4e-9831-ba3c52a06518)

The cool thing is that Maven and Gradle consumers will use both the very same coordinate, but each of them will actually consume only its corresponding system: `pom.xml` for Maven, `module.json` for Gradle.

I just tried/tested this locally within Gradle only, so, please, give me a feedback whether this is working also from Maven on your CI

Commit:
Flat versions, quick fix for Gradle consumers #81

Commit:
add comment explaining the flattening

Commit:
injecting Gradle Platform and Catalog within the same Maven publication

cowtowncoder commented on .gitattributes:
```diff
@@ -1,7 +1,6 @@
+#
+# These are explicitly windows files and should use crlf
+*.bat           text eol=crlf
 
```
Please do not overwrite existing settings -- `pom.xml` conflicts are a big nuisance across branches.

cowtowncoder commented:
Ok, I am bit confused here. So, I thought this would be the alternative to flattening versions... but looks like PR also contains flattening changes?

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle

cowtowncoder commented:
Looks like CI wasn't enabled for PRs (haven't gotten too many), enabled now.

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle

elect86 commented:
> Ok, I am bit confused here. So, I thought this would be the alternative to flattening versions... but looks like PR also contains flattening changes?

Sorry, this was supposed to take some time before getting it to work (since the one in Scijava lasted quite long), so I prepared this in the meanwhile, while the other one could be merged right away

However I should have switched back branch before creating a new one, let's close this one and create a new one

Commit:
Revert "Flat versions, quick fix for Gradle consumers #81"

This reverts commit 77da4b4807fc4b139f1cf5e897528d37aed7866f.

Commit:
Merge remote-tracking branch 'origin/2.17-gradle' into 2.17-gradle

Commit:
revert accidental .gitattributes changes

Commit:
revert accidental .gitattributes changes

Commit:
remove the pom.xml comment

elect86 commented:
Ok, it should be back on track now

Sorry for all the noise..

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle

cowtowncoder commented on pom.xml:
```diff
@@ -442,6 +443,90 @@ of Jackson components maintained by FasterXML.com
+        </file>
+      </activation>
+      <build>
+        <plugins>
```
Would be nice to have brief Javadoc referencing this PR, explaining what is being done.

cowtowncoder commented:
CI fails (looks like I need to manually approve since it's your first contribution).

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle

elect86 commented:
Sorry for the late reply

I might write some brief javadoc, if still interested

> need to specify plug-in version(s).

Also, is this still relevant?

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle

cowtowncoder commented:
> Sorry for the late reply
> 
> I might write some brief javadoc, if still interested
> 
> > need to specify plug-in version(s).
> 
> Also, is this still relevant?

Yes. You haven't tried Maven build for the project? It fails on CI -- see `Checks`.

Locally fails for me with

    ./mvnw clean compile

elect86 committed:
Merge branch 'FasterXML:2.19' into 2.17-gradle

elect86 commented:
> Yes. You haven't tried Maven build for the project? It fails on CI -- see `Checks`.

No sorry, I didn't pay attention, but yeah, you are right

However `mvnw clean compile` runs fine on mine, weird

> elect@gibanb02:~/IdeaProjects/jackson-bom$ ./mvnw clean compile
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 2 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Jackson BOM                                                        [pom]
[INFO] Jackson Base                                                       [pom]
[INFO] 
[INFO] -----------------< com.fasterxml.jackson:jackson-bom >------------------
[INFO] Building Jackson BOM 2.19.0-SNAPSHOT                               [1/2]
[INFO]   from pom.xml
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- clean:3.4.1:clean (default-clean) @ jackson-bom ---
[INFO] 
[INFO] --- enforcer:3.5.0:enforce (enforce-java) @ jackson-bom ---
[INFO] Rule 0: org.apache.maven.enforcer.rules.version.RequireJavaVersion passed
[INFO] Rule 1: org.apache.maven.enforcer.rules.version.RequireMavenVersion passed
[INFO] Rule 2: org.apache.maven.enforcer.rules.RequirePluginVersions passed
[INFO] 
[INFO] --- build-helper:3.6.0:add-source (add-generated-sources) @ jackson-bom ---
[INFO] Source directory: /home/elect/IdeaProjects/jackson-bom/target/generated-sources added.
[INFO] 
[INFO] -----------------< com.fasterxml.jackson:jackson-base >-----------------
[INFO] Building Jackson Base 2.19.0-SNAPSHOT                              [2/2]
[INFO]   from base/pom.xml
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- clean:3.4.1:clean (default-clean) @ jackson-base ---
[INFO] 
[INFO] --- build-helper:3.6.0:regex-property (regex-property) @ jackson-base ---
[INFO] 
[INFO] --- enforcer:3.5.0:enforce (enforce-java) @ jackson-base ---
[INFO] Rule 0: org.apache.maven.enforcer.rules.version.RequireJavaVersion passed
[INFO] Rule 1: org.apache.maven.enforcer.rules.version.RequireMavenVersion passed
[INFO] Rule 2: org.apache.maven.enforcer.rules.RequirePluginVersions passed
[INFO] 
[INFO] --- build-helper:3.6.0:add-source (add-generated-sources) @ jackson-base ---
[INFO] Source directory: /home/elect/IdeaProjects/jackson-bom/base/target/generated-sources added.
[INFO] 
[INFO] --- build-helper:3.6.0:add-resource (add-resource) @ jackson-base ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Jackson BOM 2.19.0-SNAPSHOT:
[INFO] 
[INFO] Jackson BOM ........................................ SUCCESS [  0.195 s]
[INFO] Jackson Base ....................................... SUCCESS [  0.103 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.196 s
[INFO] Finished at: 2025-04-23T15:40:39+02:00
[INFO] ------------------------------------------------------------------------


Anyway, I just pushed a commit with the plugins versions (I picked the latest ones, tell me if this may not be fine for whatever reason)

Commit:
Added missing version specifications for Maven plugins

cowtowncoder commented:
> However mvnw clean compile runs fine on mine, weird

Often that'd be due to older version of Maven (at some point things changed to stricter defaults), but if using `mvnw` from repo that should not be the case. Oh well.

cowtowncoder commented:
@elect86 Latest versions should be fine but... for some reason, CI now just fails. Hmmh.

Commit:
Update Gradle wrapper to version 8.13

Moved and upgraded the Gradle wrapper configuration from 8.12.1 to 8.13. The wrapper files were in the wrong directory.

elect86 commented:
Sorry, my bad, dummy mistake, gradle wrapper is supposed to be found under the `gradle` directory from the `gradlew` point of view.. now it should run

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle

cowtowncoder committed:
Merge branch '2.19' into 2.17-gradle
